framework/db/Connection.php

```diff
@@ -607,6 +611,7 @@ protected function initConnection()
         if ($this->charset !== null && in_array($this->getDriverName(), ['pgsql', 'mysql', 'mysqli', 'cubrid'])) {
             $this->pdo->exec('SET NAMES ' . $this->pdo->quote($this->charset));
         }
+        $this->getSupportsResultSets();
```

Comment:
I don't know really why, but let's assume you have error in multi-query and expect exception ("SQLSTATE[42S22]: Column not found: ..."). Commenting out `$this->getSupportsResultSets();` in `Connection` will then throw the exception here:

```php
$pdoStatement = $this->pdo->prepare(''); // we create new PDOStatement instance
$pdoStatement->nextRowSet(); <------------------------------
```
instead of here

```php
while ($this->pdoStatement->nextRowSet()) {}
```

While it's obviously not the same PDOStatement instance, internally something strange is going on.

So only solution I see is to do this stuff as soon as connection is opened.